crates/ruma-state-res/src/event_auth.rs

```diff
@@ -711,11 +711,13 @@ fn valid_membership_change(
                         "Can't make another user join, sender did not match target"
                     );
                     false
-                } else if matches!(sender_membership, MembershipState::Ban | MembershipState::Join)
-                {
+                } else if matches!(
+                    sender_membership,
+                    MembershipState::Ban | MembershipState::Invite | MembershipState::Join
+                ) {
                     warn!(
                         ?target_user_membership_event_id,
-                        "Membership state of ban or join are invalid",
+                        "Membership state of ban, invite or join are invalid",
                     );
                     false
                 } else {
```

Comment:
Might be nice to turn these two branches into one else + exhaustive match on `sender_membership`, so it's clearer what cases lead to `true`. Can be a separate PR though.

Reply:
Done in #1736